solidity/dashboard/src/components/Timeline.jsx

```diff
@@ -0,0 +1,51 @@
+import React, { useContext } from "react"
+
+const TimelineContext = React.createContext({
+  // May be useful in the future, eg. if we would like to be able to display the
+  // timeline events on alternating sides.
+  position: "left",
+})
+
+const useTimelineContext = () => {
+  const context = useContext(TimelineContext)
+  if (!context) {
+    throw new Error("TimelineContext used outside of Timeline component")
+  }
+  return context
+}
+
+const Timeline = ({ children, position = "left" }) => {
+  return (
+    <TimelineContext.Provider value={{ position: position }}>
+      <div className="timeline">{children}</div>
+    </TimelineContext.Provider>
+  )
+}
+const TimelineElement = ({ position, children }) => {
+  const { position: timelinePosition } = useTimelineContext()
+  const _position = position || timelinePosition
+  return (
+    <div className={`timeline__element timeline__element--${_position}`}>
```

Comment:
I think we should get used to writing a classNames as property when creating new components and adding it immediately to (preferably) every div possible 😅  This is some light idea that would be great to discuss, but it could make the components even more reusable.

We could just keep it in one property as object (e.g classes) and assign them to the divs inside components.

For example:
```
const TimelineElement = ({ position, **classes**, children }) => {
  const { position: timelinePosition } = useTimelineContext()
  const _position = position || timelinePosition
  return (
    <div className={`timeline__element timeline__element--${_position} ${classes.root ? classes.root : ''}`}>
      {children}
    </div>
  )
}
```
If later we add the component title, then we do `${classes.title ? classes.title : ''}`.

And when using the components we will just write:
```
<TimelineElement classes={{
  root: 'root-class-name',
  title: 'title-class-name',
}} />
```

This idea is actually something i got from `Angular Material` when they use the classes similarly like here.


So, not saying that you have to do it like that, but I'm interesting in your opinion about this solution.

Reply:
Also the background of it on Figma is violet, but we could actually specific it with the added class like i mentioned above and keep it with white background as default.

Reply:
> Also the background of it on Figma is violet, but we could actually specific it with the added class like i mentioned above and keep it with white background as default.

I believe the parent component should define the background. So in this case the section tag should set the background. eg

```
<section className="tile **bg-vilolet**">
 <Timeline>
  ...
 </Timeline>
</section>
```

Reply:
IMO (ofc I'm open to discussion 😆) passing `classes` props only make sense in non-compound style components eg.:
```
const Component = ({ classes, title, subtitle, description, }) => {
  <section className={`default-style ${classes.root}`}>
    <h1 className={`default-title-style ${classes.title}`}>{title}</h1>
    <h3 className={`default-subtitle-style ${classes.subtitle}`}>{subtitle}</h3>
    <span className={`default-description-style ${classes.description}`}>
      {description}
    </span>
  </section>
}
```

In compound components, we are able to add a custom class name that can override default styles. Thoughts?

I missed the `className` prop for the `Timeline.Element` component- I'll update.

Reply:
Regarding background: I temporarily added the `Timeline` just to show how it looks like. I'll create the right background and component grid in #2543.